### ISSUE TITLE
plot(section, which="temperature") ought to show in-situ temperature

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1899]
### Summary

@clayton33 and @richardsc -- this is a heads-up about likely changes to `plot,section-method`.  The issue boils down to a desire to reduce possible conflicts between the `which` and `eos` values.  I think the existing code has an error, because the label for `which="temperature"` depends on the `eos` value, which makes no sense.  I think `"temperature"` must mean in-situ temperature, which is *not* dependent on EOS.  Whether that's what is being plotted or not, for sure the label changes with the `eos` value, which makes no sense.  So, we have an error I need to fix.  Frankly, I am thinking that the `eos` argument is simply a cause for trouble.

### Changes I think are required

1. Let users plot anything in the data, or that can be derived from the data. This is an expansion of #1878.
2. Check that `which="temperature"` actually plots in-situ temperature.  Right now, the CRAN version labels it differently according to the `eos` value, but I have not checked on the contours yet so it might just be a labelling error.
3. Extend and clarify the documentation w.r.t. `which`, especially with regard to the `which="temperature"` case. 
4. Ignore the `eos` parameter.  
 
The last of these is what I hope @clayton33 and @richardsc will have a chance to think about.

### Background

During work for #1878 I noticed a confusion on what `which="temperature"` does.  The docs (https://dankelley.github.io/oce/reference/plot-section-method.html) say
> which=1 or "temperature" for temperature contours (the default)
but I think this is not what is actually plotted.  I think it plots a form of potential temperature.

Things are confusing, with the `eos` parameter to the function.   The intention (I think) was that `which="potential temperature"` would do one thing for `eos="unesco"` and another for `eos="gsw"`.  This might have made sense a decade ago when GSW was just starting to be known, but I think now users will know the difference, and can be relied upon to just say `"potential temperature"` to get unesco, and `"Conservative Temperature"` to get that thing.  Similarly, I think if a person asks for "salinity" then they should get the unesco form.

I am thinking "out loud" here, but the point of this is so co-developers will see that some changes may be coming.

Since (I think) the behaviour of the CRAN version does not match the docs, my plan is to (a) change the code to obey the docs and (b) clarify those docs.  Whether it makes sense to keep the `eos` argument is very much in question.  Keeping it complicates the code quite a lot, and I think will confuse the heck out of users.  For an example, if a user writes
```R
plot(section, which="SA", eos="unesco")
```
then what  the *heck* do they want??? We'd have to make a choice in the code, which is just messy to code and complex to document.  And, what's the point?  At this stage, oceanographers ought to know that `"SA"` refers to `"Absolute Salinity"`.  As I am extending the plot method to let users plot pretty much anything they want, it's time to clarify the functionality.

**Comment by dankelley** [comment]
I can verify that the CRAN version with
```R
plot(section, which="temperature")
```
labels with upper-case Theta (indicating the GSW Conservative Temperature) but that it is actually plotting the in-situ temperature.  Thus, the label is incorrect, but the existing code is matching the documentation.

Therefore, I will change the label.

My apologies, folks.  This has been incorrect for some time.  I will note the change in the NEWS file, since it's important to users.

<details>

*Snapshot with old code*
<img width="793" alt="001_temperature_old" src="https://user-images.githubusercontent.com/99469/147767494-0abc5b14-e92c-408e-b48f-44aee446e7c1.png">


*Snapshot with new code*
<img width="793" alt="002_temperature_new" src="https://user-images.githubusercontent.com/99469/147767503-0ffd8dcd-6373-4869-a55c-90fe4b9318b0.png">

**Comment by richardsc** [comment]
I'm on the road, so can't dig into this too deep, but from your description of the error and the issues I agree with what you have proposed, specifically to remove the eos from `plot,section-method`

**Comment by dankelley** [comment]
Actually, `eos` is required because e.g. `sigma1` is formulated differently in the gsw and unesco systems.

I am going to go ahead and close this issue, because the matter (as stated in the title) has been addressed in commit bbaadfa9b8a52d1f6435d878eaa6d512f4a14e66  of the "develop" branch.
